lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -30,24 +30,24 @@ namespace Org.Apache.REEF.Common.Runtime.Evaluator.Task
     internal sealed class TaskStatus
     {
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(TaskStatus));
+
+        private readonly object _stateLock = new object();
         private readonly TaskLifeCycle _taskLifeCycle;
         private readonly HeartBeatManager _heartBeatManager;
         private readonly Optional<ISet<ITaskMessageSource>> _evaluatorMessageSources;
 
-        private readonly string _taskId;
-        private readonly string _contextId;
         private Optional<Exception> _lastException = Optional<Exception>.Empty();
         private Optional<byte[]> _result = Optional<byte[]>.Empty();
         private TaskState _state;
 
         public TaskStatus(HeartBeatManager heartBeatManager, string contextId, string taskId, Optional<ISet<ITaskMessageSource>> evaluatorMessageSources)
         {
-            _contextId = contextId;
-            _taskId = taskId;
             _heartBeatManager = heartBeatManager;
             _taskLifeCycle = new TaskLifeCycle();
             _evaluatorMessageSources = evaluatorMessageSources;
             State = TaskState.Init;
+            TaskId = taskId;
```

Comment:
Doesn't the use of a private setter instead of a readonly field make things less safe?